File: Programacion2/POO/Lab06_2023-2/Bibliotecas/Cliente.cpp
#include <cstring>
#include <fstream>
using namespace std;

#include "Cliente.h"
Cliente::Cliente() {
    nombre = nullptr;
    dni = 0;
    telefono = 0;
    cantidad_productos_entregados = 0;
    monto_total = 0;
}
Cliente::~Cliente() {
    if (nombre) delete[] nombre;
    dni = 0;
    telefono = 0;
    cantidad_productos_entregados = 0;
    monto_total = 0;
}
void Cliente::setDni(int dni) {
    this->dni = dni;
}
int Cliente::getDni() const {
    return dni;
}
void Cliente::setNombre(char* nomb) {
    if (nombre) delete[] nombre;
    nombre = new char[strlen(nomb)+1];
    strcpy(nombre, nomb);
}
void Cliente::getNombre(char* nomb) const {
    if (nombre == nullptr) nomb[0] = 0;
    else strcpy(nomb, nombre);
}
void Cliente::setTelefono(int tel) {
    telefono = tel;
}
int Cliente::getTelefono() const {
    return telefono;
}
void Cliente::setMontoTotal(double monto) {
    monto_total = monto;
}
double Cliente::getMontoTotal() const {
    return monto_total;
}

void Cliente::operator+=(Pedido& pedido) {

}

bool operator>>(ifstream& arch, Cliente& cliente) {
    // 79464412,PORTUGAL RAFFO ALEXANDER,3902394
    // dni, nombre, telefono
    int dni, telefono;
    char nombre[100];

    arch >> dni;
    if (arch.eof()) return false;
    arch.get();
    arch.getline(nombre, 100, ',');
    arch >> telefono;

    cliente.setDni(dni);
    cliente.setTelefono(telefono);
    cliente.setNombre(nombre);
    cliente.setMontoTotal(0.0);

    return true;
}

File: Programacion2/Java/Lab10_2024-1/src/Cliente.java
import java.util.Scanner;

public class Cliente {
    private int dni;
    private String nombre;
    private double totalGastado;

    public Cliente() {
        totalGastado = 0;
    }

    public int getDni() {
        return dni;
    }

    public void setDni(int dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getTotalGastado() {
        return totalGastado;
    }

    public void setTotalGastado(double totalGastado) {
        this.totalGastado = totalGastado;
    }

    public boolean leer(Scanner arch){
        dni = arch.nextInt();
        if (dni == 0) return false;

        nombre = arch.next();

        return true;
    }
}
